newIDE/app/src/MainFrame/index.js

```diff
@@ -1889,13 +1889,23 @@ class MainFrame extends React.Component<Props, State> {
           (resourceSource, index): React.Node => {
             const Component = resourceSource.component;
             return (
-              <Component
-                key={resourceSource.name}
-                ref={dialog =>
-                  (this._resourceSourceDialogs[resourceSource.name] = dialog)
-                }
-                i18n={i18n}
-              />
+              <PreferencesContext.Consumer>
+                {({ loadLatestPath, saveLatestPath }) => {
+                  return (
+                    <Component
+                      key={resourceSource.name}
+                      ref={dialog =>
+                        (this._resourceSourceDialogs[
+                          resourceSource.name
+                        ] = dialog)
+                      }
+                      i18n={i18n}
+                      loadLatestPath={loadLatestPath}
```

Comment:
Name this function `getLatestPath` or `getLastUsedPath`

(load => get, because there is no loading necessary)